javaslang/src/main/java/javaslang/collection/Array.java

```diff
@@ -1048,11 +1048,15 @@ public int length() {
 
     @Override
     public Array<T> reverse() {
-        final Object[] arr = new Object[delegate.length];
-        for (int i = 0; i < delegate.length; i++) {
-            arr[delegate.length - 1 - i] = delegate[i];
+        if (size() <= 1) {
+            return this;
+        } else {
+            final Object[] arr = new Object[delegate.length];
+            for (int i = 0; i < delegate.length; i++) {
+                arr[delegate.length - 1 - i] = delegate[i];
+            }
```

Comment:
@paplorinc I not completely agree. The are things defined by the JVM spec that are crystal clear, e.g. heap vs. stack access.  Stack access can't be faster (other than register access), only heap access can be faster. There are two usages of `delegate.length`, which is kind of dedundant. Adding a local length variable also clarifies that the same length is meant.

Regarding the existing issues I would add a priority $numberOfIssues (greater is less important) to adding a benchmark for the use of a local variable.